libr/include/r_cons.h

```diff
@@ -878,8 +879,13 @@ typedef int (*RLineCallback)(RLine *line);
 
 typedef struct r_line_comp_t {
 	bool opt;
+
+	// private! use r_line_completion_argc(), r_line_completion_argv()
 	int argc;
-	const char **argv;
+	const char **argv; // weak reference to a whole list, if NULL, args will be used
```

Comment:
uhm isnt args replacing argc+argv ?

Reply:
args replaces argc+argv iff `argv == NULL`. This argv is for directly setting a whole list, like in `r_core_fgets()` without having to copy anything.